platform-api/che-core-api-auth/src/main/java/org/eclipse/che/security/oauth/OAuthAuthenticationService.java

```diff
@@ -236,6 +240,23 @@ public Response invalidate() throws IOException {
                                                                            null)).type(MediaType.TEXT_PLAIN).build();
     }
 
+    /**
+     * Gets OAuth token for user.
+     *
+     * @param oauthProvider OAuth provider name
+     * @return OAuthToken
+     * @throws IOException
+     */
+    @GET
+    @Path("token")
+    @Produces(MediaType.APPLICATION_JSON)
+    @RolesAllowed({"user", "temp_user"})
+    public OAuthToken token(@Required @QueryParam("oauth_provider") String oauthProvider)
+            throws IOException {
+        OAuthAuthenticator provider = getAuthenticator(oauthProvider);
```

Comment:
We are not using WebApplicationException anymore. Please replace it with BadRequestException